x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobConfig.java

```diff
@@ -56,15 +56,14 @@ public String getCron() {
         return "*";
     }
 
-    public void writeTo(StreamOutput out) throws IOException {
+    public void writeTo(final StreamOutput out) throws IOException {
         out.writeString(id);
     }
 
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+    public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
         builder.startObject();
-        if (id != null) {
-            // to be replace by constant
-            builder.field("id", id);
+        {
+            builder.field(ID, id);
         }
```

Comment:
Why are you keeping the scope if there is no longer a branch that prevents entry?

Reply:
I agree, took this over from another place and was wondering myself, but then thought: 

As this builds a json object it emulates the json structure. So this is rather syntactic sugar: forcing indentation to make the code more readable. 

(Right now the whole config is bogus, because it is empty. But this will be the next step.